src/promptflow/promptflow/_cli/_pf/_flow.py

```diff
@@ -423,19 +426,55 @@ def test_flow(args):
 
 
 def serve_flow(args):
+    from promptflow._sdk._load_functions import load_flow
+
     logger.info("Start serve model: %s", args.source)
     # Set environment variable for local test
     source = Path(args.source)
+    logger.info(
+        "Start promptflow server with port %s",
+        args.port,
+    )
     os.environ["PROMPTFLOW_PROJECT_PATH"] = source.absolute().as_posix()
-    from promptflow._sdk._serving.app import create_app
+    flow = load_flow(args.source)
+    if "language" in flow.dag and flow.dag["language"] == FlowLanguage.CSharp:
+        serve_flow_csharp(args, source)
+    else:
+        serve_flow_python(args, source)
+    logger.info("Promptflow app ended")
+
 
+def serve_flow_csharp(args, source):
+    from promptflow.batch._csharp_executor_proxy import EXECUTOR_SERVICE_DLL
+    try:
+        # Change working directory to model dir
+        print(f"Change working directory to model dir {source}")
+        os.chdir(source)
+        command = [
+            "dotnet",
+            EXECUTOR_SERVICE_DLL,
+            "-p",
+            str(args.port),
+            "-y",
+            "flow.dag.yaml",
+            "-a",
+            ".",
+            "-c",
```

Comment:
please use full parameter name here to let people know the meaning of these magic default value

Reply:
updated